app/controllers/bookings_controller.rb

```diff
@@ -1,21 +1,24 @@
 class BookingsController < ApplicationController
   before_action except: [:view_booking, :paypal_hook] do
+    authenticate_user
     set_event
     ticket_params
+    is_ticket_quantity_specified
   end
   protect_from_forgery except: [:paypal_hook]
 
+
   def create
     @booking = Booking.create(event: @event, user: current_user)
 
     tickets = []
     ticket_params.each{ |ticket_type_id, quantity|
-        quantity.to_i.times{
+        quantity.to_i.times{ |i|
```

Comment:
Unused block argument - `i`. You can omit the argument if you don't care about it.<br>Avoid using {...} for multi-line blocks.<br>Use 2 (not 4) spaces for indentation.